### ISSUE TITLE
[1.15.2]GuiScreenEvent.MouseClickedEvent.Post not fired for ContainerScreen

### ISSUE THREAD

**Issue by GotoLink** [MinecraftForge/MinecraftForge#6060]
**Minecraft Version:** 1.15.2

**Forge Version:** 31.1.35

**Logs:** {Link(s) to GitHub Gist with full latest.log and/or crash report}

**Steps to Reproduce:**
 1. Record a listener for this specific case:
`private void clickInContainer(GuiScreenEvent.MouseClickedEvent.Post event){
System.out.println("Mouse click");
}`
 2. Open a workbench or furnace or whatever block inventory
 3. Click anywhere

**Description of issue:**
I used this event in 1.12 for handling shift clicks on slots for various crafting screens (vanilla and modded ones). I prefer the post action event to have an accurate state of the inventory contents, and thus minimize conflict.
From what i can tell ContainerScreen#mouseClicked(double,double,int) always returns true, meaning the patch in MouseHelper is without effect.

**Comment by reserveword** [comment]
also exists on 30.0.16.

**Comment by GotoLink** [comment]
Update for 1.15.2.
Also applies to MouseReleasedEvent.Post

**Comment by Shadows-of-Fire** [comment]
[The old `MouseInputEvent` unconditionally fired `Post`](https://github.com/MinecraftForge/MinecraftForge/blob/1.12.x/patches/minecraft/net/minecraft/client/gui/GuiScreen.java.patch#L94-L97) unless it was explicitly blocked by the GUI - the [new event that was implemented in 1.13](https://github.com/MinecraftForge/MinecraftForge/commit/b4cf6f7f364796431267f5ef662a6663df683a29#diff-c84e03230d08e607c20147d4ede211a90f1e47837be938b64a07338bee4b7267R7) does not share the same logic chain as the old one. 

 The current logic chain unconditionally fires `Pre`, and if `Pre` is cancelled it skips calls to `mouseClicked` and `Post`.
However, if `mouseClicked` returns true (which indicates the click has been handled by a Gui component), then `Post` does not fire.

A PR should be made that ideally changes the control flow of `MouseClickedEvent` and `MouseReleasedEvent` to the following
```
Pre - always fires, if cancelled the execution chain stops, indicates a click is starting
<vanilla click handler executes and returns a result>
Post - fires to indicate the click has been passed through the gui, regardless of the gui's handling (but it should receive the gui handled result)
```
The changes needed to implement this are non-breaking.
